homeassistant/components/mqtt/cover.py

```diff
@@ -401,11 +402,12 @@ def supported_features(self):
 
         return supported_features
 
-    async def async_open_cover(self, **kwargs):
+    def open_cover(self, **kwargs):
         """Move the cover up.
 
         This method is a coroutine.
         """
+        self._state = STATE_OPENING
```

Comment:
Don't set it here, MQTT cover is not always optimistic.
We should still rely on state updates either through received MQTT messages or through optimistic state updates after a service call.

Reply:
Then there is a Bug somewhere ... if i add this "live state" in the gui (and entity state) i can really see what the cover is doing. The "moving state" (opening and closing" is never set anywhere else

Reply:
If the entity is non-optimistic, i.e. it will respect (and expect) state updates received from the device over MQTT, the opening/closing states are set in `state_message_received`.
If the entity is not optimistic there's no way to know what the device is doing, so the state will just be set to whatevver MQTT message was last sent, i.e. either `STATE_OPEN` or `STATE_CLOSED`.

Reply:
Ok i will check that, and why Shelly devices are not using this.

Reply:
This is possibly related #43949, in case the Shellies report absolute position?

Reply:
You impress me again and again ... yes that seems to be the problem. As i expected the bug is somewhere else!

Im removing this!
But the toggle function is only working if state is correct (when to execute stop) so this needs to be solved and merged before mine.